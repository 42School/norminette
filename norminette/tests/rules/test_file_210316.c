int	main(void)
{
	if (((unsigned char*)s1)[curr] != ((unsigned char*)s2)[curr])
		return (((unsigned char*)s1)[curr] - ((unsigned char*)s2)[curr]);
}

struct my_struct			g_struct = {
	.field = (void *)&value
};

int	main(void)
{
	t_quaternion	g_quat = {
		.r = 1,
		.i = 0,
		.j = 0,
		.k = 0
	};
}
t_quaternion				g_quat = {
	.r = 1,
	.i = 0,
	.j = 0,
	.k = 0
};
*result = (t_quaternion)
{
	.r = 1,
	.i = 0,
	.j = 0,
	.k = 0
};

static const t_quaternion	g_quat = {.r = 1, .i = 0, .j = 0, .k = 0};
static const t_quaternion	g_quat = {r : 1, i : 0, j: 0, k : 0 };
static const t_quaternion	g_quats[4] = {[0].r = 3, [1].i = 5 };
*result = (t_quaternion){1, 0, 0, 0};
*result = (t_quaternion){r : 1, i : 0, j : 0, k : 0 };