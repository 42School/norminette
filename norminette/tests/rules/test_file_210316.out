[36mtest_file_210316.c[0m - [32mIsFuncDeclaration[0m In "GlobalScope" from "None" line 1":
		<INT> <TAB> <IDENTIFIER=main> <LPARENTHESIS> <VOID> <RPARENTHESIS> <NEWLINE>
[36mtest_file_210316.c[0m - [32mIsBlockStart[0m In "Function" from "GlobalScope" line 2":
		<LBRACE> <NEWLINE>
[36mtest_file_210316.c[0m - [32mIsControlStatement[0m In "Function" from "GlobalScope" line 3":
		<TAB> <IF> <SPACE> <LPARENTHESIS> <LPARENTHESIS> <LPARENTHESIS> <UNSIGNED> <SPACE> <CHAR> <MULT> <RPARENTHESIS> <IDENTIFIER=s1> <RPARENTHESIS> <LBRACKET> <IDENTIFIER=curr> <RBRACKET> <SPACE> <NOT_EQUAL> <SPACE> <LPARENTHESIS> <LPARENTHESIS> <UNSIGNED> <SPACE> <CHAR> <MULT> <RPARENTHESIS> <IDENTIFIER=s2> <RPARENTHESIS> <LBRACKET> <IDENTIFIER=curr> <RBRACKET> <RPARENTHESIS> <NEWLINE>
[36mtest_file_210316.c[0m - [32mIsExpressionStatement[0m In "ControlStructure" from "Function" line 4":
		<TAB> <TAB> <RETURN> <SPACE> <LPARENTHESIS> <LPARENTHESIS> <LPARENTHESIS> <UNSIGNED> <SPACE> <CHAR> <MULT> <RPARENTHESIS> <IDENTIFIER=s1> <RPARENTHESIS> <LBRACKET> <IDENTIFIER=curr> <RBRACKET> <SPACE> <MINUS> <SPACE> <LPARENTHESIS> <LPARENTHESIS> <UNSIGNED> <SPACE> <CHAR> <MULT> <RPARENTHESIS> <IDENTIFIER=s2> <RPARENTHESIS> <LBRACKET> <IDENTIFIER=curr> <RBRACKET> <RPARENTHESIS> <SEMI_COLON> <NEWLINE>
[36mtest_file_210316.c[0m - [32mIsBlockEnd[0m In "Function" from "GlobalScope" line 5":
		<RBRACE> <NEWLINE>
[36mtest_file_210316.c[0m - [32mIsEmptyLine[0m In "GlobalScope" from "None" line 6":
		<NEWLINE>
[36mtest_file_210316.c[0m - [32mIsVarDeclaration[0m In "GlobalScope" from "None" line 7":
		<STRUCT> <SPACE> <IDENTIFIER=my_struct> <TAB> <TAB> <TAB> <IDENTIFIER=g_struct> <SPACE> <ASSIGN> <SPACE> <LBRACE> <NEWLINE>
		<TAB> <DOT> <IDENTIFIER=field> <SPACE> <ASSIGN> <SPACE> <LPARENTHESIS> <VOID> <SPACE> <MULT> <RPARENTHESIS> <BWISE_AND> <IDENTIFIER=value> <NEWLINE>
		<RBRACE> <SEMI_COLON> <NEWLINE>
[36mtest_file_210316.c[0m - [32mIsEmptyLine[0m In "GlobalScope" from "None" line 10":
		<NEWLINE>
[36mtest_file_210316.c[0m - [32mIsFuncDeclaration[0m In "GlobalScope" from "None" line 11":
		<INT> <TAB> <IDENTIFIER=main> <LPARENTHESIS> <VOID> <RPARENTHESIS> <NEWLINE>
[36mtest_file_210316.c[0m - [32mIsBlockStart[0m In "Function" from "GlobalScope" line 12":
		<LBRACE> <NEWLINE>
[36mtest_file_210316.c[0m - [32mIsVarDeclaration[0m In "Function" from "GlobalScope" line 13":
		<TAB> <IDENTIFIER=t_quaternion> <TAB> <IDENTIFIER=g_quat> <SPACE> <ASSIGN> <SPACE> <LBRACE> <NEWLINE>
		<TAB> <TAB> <DOT> <IDENTIFIER=r> <SPACE> <ASSIGN> <SPACE> <CONSTANT=1> <COMMA> <NEWLINE>
		<TAB> <TAB> <DOT> <IDENTIFIER=i> <SPACE> <ASSIGN> <SPACE> <CONSTANT=0> <COMMA> <NEWLINE>
		<TAB> <TAB> <DOT> <IDENTIFIER=j> <SPACE> <ASSIGN> <SPACE> <CONSTANT=0> <COMMA> <NEWLINE>
		<TAB> <TAB> <DOT> <IDENTIFIER=k> <SPACE> <ASSIGN> <SPACE> <CONSTANT=0> <NEWLINE>
		<TAB> <RBRACE> <SEMI_COLON> <NEWLINE>
[36mtest_file_210316.c[0m - [32mIsBlockEnd[0m In "Function" from "GlobalScope" line 19":
		<RBRACE> <NEWLINE>
[36mtest_file_210316.c[0m - [32mIsVarDeclaration[0m In "GlobalScope" from "None" line 20":
		<IDENTIFIER=t_quaternion> <TAB> <TAB> <TAB> <TAB> <IDENTIFIER=g_quat> <SPACE> <ASSIGN> <SPACE> <LBRACE> <NEWLINE>
		<TAB> <DOT> <IDENTIFIER=r> <SPACE> <ASSIGN> <SPACE> <CONSTANT=1> <COMMA> <NEWLINE>
		<TAB> <DOT> <IDENTIFIER=i> <SPACE> <ASSIGN> <SPACE> <CONSTANT=0> <COMMA> <NEWLINE>
		<TAB> <DOT> <IDENTIFIER=j> <SPACE> <ASSIGN> <SPACE> <CONSTANT=0> <COMMA> <NEWLINE>
		<TAB> <DOT> <IDENTIFIER=k> <SPACE> <ASSIGN> <SPACE> <CONSTANT=0> <NEWLINE>
		<RBRACE> <SEMI_COLON> <NEWLINE>
[36mtest_file_210316.c[0m - [32mIsAssignation[0m In "GlobalScope" from "None" line 26":
		<MULT> <IDENTIFIER=result> <SPACE> <ASSIGN> <SPACE> <LPARENTHESIS> <IDENTIFIER=t_quaternion> <RPARENTHESIS> <NEWLINE>
		<LBRACE> <NEWLINE>
		<TAB> <DOT> <IDENTIFIER=r> <SPACE> <ASSIGN> <SPACE> <CONSTANT=1> <COMMA> <NEWLINE>
		<TAB> <DOT> <IDENTIFIER=i> <SPACE> <ASSIGN> <SPACE> <CONSTANT=0> <COMMA> <NEWLINE>
		<TAB> <DOT> <IDENTIFIER=j> <SPACE> <ASSIGN> <SPACE> <CONSTANT=0> <COMMA> <NEWLINE>
		<TAB> <DOT> <IDENTIFIER=k> <SPACE> <ASSIGN> <SPACE> <CONSTANT=0> <NEWLINE>
		<RBRACE> <SEMI_COLON> <NEWLINE>
[36mtest_file_210316.c[0m - [32mIsEmptyLine[0m In "GlobalScope" from "None" line 33":
		<NEWLINE>
[36mtest_file_210316.c[0m - [32mIsVarDeclaration[0m In "GlobalScope" from "None" line 34":
		<STATIC> <SPACE> <CONST> <SPACE> <IDENTIFIER=t_quaternion> <TAB> <IDENTIFIER=g_quat> <SPACE> <ASSIGN> <SPACE> <LBRACE> <DOT> <IDENTIFIER=r> <SPACE> <ASSIGN> <SPACE> <CONSTANT=1> <COMMA> <SPACE> <DOT> <IDENTIFIER=i> <SPACE> <ASSIGN> <SPACE> <CONSTANT=0> <COMMA> <SPACE> <DOT> <IDENTIFIER=j> <SPACE> <ASSIGN> <SPACE> <CONSTANT=0> <COMMA> <SPACE> <DOT> <IDENTIFIER=k> <SPACE> <ASSIGN> <SPACE> <CONSTANT=0> <RBRACE> <SEMI_COLON> <NEWLINE>
[36mtest_file_210316.c[0m - [32mIsVarDeclaration[0m In "GlobalScope" from "None" line 35":
		<STATIC> <SPACE> <CONST> <SPACE> <IDENTIFIER=t_quaternion> <TAB> <IDENTIFIER=g_quat> <SPACE> <ASSIGN> <SPACE> <LBRACE> <IDENTIFIER=r> <SPACE> <COLON> <SPACE> <CONSTANT=1> <COMMA> <SPACE> <IDENTIFIER=i> <SPACE> <COLON> <SPACE> <CONSTANT=0> <COMMA> <SPACE> <IDENTIFIER=j> <COLON> <SPACE> <CONSTANT=0> <COMMA> <SPACE> <IDENTIFIER=k> <SPACE> <COLON> <SPACE> <CONSTANT=0> <SPACE> <RBRACE> <SEMI_COLON> <NEWLINE>
[36mtest_file_210316.c[0m - [32mIsVarDeclaration[0m In "GlobalScope" from "None" line 36":
		<STATIC> <SPACE> <CONST> <SPACE> <IDENTIFIER=t_quaternion> <TAB> <IDENTIFIER=g_quats> <LBRACKET> <CONSTANT=4> <RBRACKET> <SPACE> <ASSIGN> <SPACE> <LBRACE> <LBRACKET> <CONSTANT=0> <RBRACKET> <DOT> <IDENTIFIER=r> <SPACE> <ASSIGN> <SPACE> <CONSTANT=3> <COMMA> <SPACE> <LBRACKET> <CONSTANT=1> <RBRACKET> <DOT> <IDENTIFIER=i> <SPACE> <ASSIGN> <SPACE> <CONSTANT=5> <SPACE> <RBRACE> <SEMI_COLON> <NEWLINE>
[36mtest_file_210316.c[0m - [32mIsAssignation[0m In "GlobalScope" from "None" line 37":
		<MULT> <IDENTIFIER=result> <SPACE> <ASSIGN> <SPACE> <LPARENTHESIS> <IDENTIFIER=t_quaternion> <RPARENTHESIS> <LBRACE> <CONSTANT=1> <COMMA> <SPACE> <CONSTANT=0> <COMMA> <SPACE> <CONSTANT=0> <COMMA> <SPACE> <CONSTANT=0> <RBRACE> <SEMI_COLON> <NEWLINE>
[36mtest_file_210316.c[0m - [32mIsAssignation[0m In "GlobalScope" from "None" line 38":
		<MULT> <IDENTIFIER=result> <SPACE> <ASSIGN> <SPACE> <LPARENTHESIS> <IDENTIFIER=t_quaternion> <RPARENTHESIS> <LBRACE> <IDENTIFIER=r> <SPACE> <COLON> <SPACE> <CONSTANT=1> <COMMA> <SPACE> <IDENTIFIER=i> <SPACE> <COLON> <SPACE> <CONSTANT=0> <COMMA> <SPACE> <IDENTIFIER=j> <SPACE> <COLON> <SPACE> <CONSTANT=0> <COMMA> <SPACE> <IDENTIFIER=k> <SPACE> <COLON> <SPACE> <CONSTANT=0> <SPACE> <RBRACE> <SEMI_COLON> 
test_file_210316.c: Error!
Error: SPACE_AFTER_KW       (line:   3, col:  20):	Missing space after keyword
Error: SPACE_AFTER_KW       (line:   3, col:  50):	Missing space after keyword
Error: SPACE_AFTER_KW       (line:   4, col:  28):	Missing space after keyword
Error: SPACE_AFTER_KW       (line:   4, col:  57):	Missing space after keyword
Error: DECL_ASSIGN_LINE     (line:  13, col:  28):	Declaration and assignation on a single line
Error: DECL_ASSIGN_LINE     (line:  14, col:  12):	Declaration and assignation on a single line
Error: DECL_ASSIGN_LINE     (line:  15, col:  12):	Declaration and assignation on a single line
Error: DECL_ASSIGN_LINE     (line:  16, col:  12):	Declaration and assignation on a single line
Error: DECL_ASSIGN_LINE     (line:  17, col:  12):	Declaration and assignation on a single line
